src/lib/CRSF/CRSF.cpp

```diff
@@ -90,6 +96,50 @@ void CRSF::Begin()
     //The Reciever uses seperate rx and tx pins
 }
 
+/**
+ * Determine which switch to send next.
+ * If any switch has changed since last sent, we send the lowest index changed switch
+ * and set nextSwitchIndex to that value + 1.
+ * If no switches have changed then we send nextSwitchIndex and increment the value.
+ * For pure sequential switches, all 8 switches are part of the round-robin sequence.
+ * For hybrid switches, switch 0 is sent with every packet and the rest of the switches
+ * are in the round-robin.
+ */
+uint8_t ICACHE_RAM_ATTR CRSF::getNextSwitchIndex()
+{
+    int i;
+    int firstSwitch = 0; // sequential switches includes switch 0
+
+#if defined HYBRID_SWITCHES_8
+    firstSwitch = 1; // skip 0 since it is sent on every packet
+#endif
+
+    // look for a changed switch
+    for(i=firstSwitch; i<N_SWITCHES; i++) {
+        if (currentSwitches[i] != sentSwitches[i])
+            break;
+    }
+    // if we didn't find a changed switch, we get here with i==N_SWITCHES
+    if (i==N_SWITCHES) i = nextSwitchIndex;
+
+    // keep track of which switch to send next if there are no changed switches
+    // during the next call.
+    nextSwitchIndex = (i+1) % 8;
+
+#ifdef HYBRID_SWITCHES_8
+    // for hydrid switches 0 is sent on every packet, so we can skip
+    // that value for the round-robin
+    if (nextSwitchIndex==0)
+        nextSwitchIndex = 1;
+#endif
+
+    // since we are going to send i, we can put that value into the sent list.
```

Comment:
Fuzzy thinking. A 'get' function shouldn't side-effect state. Move this assignment out to where the switch is actually sent.

Reply:
Added a new set function to indicate the value that was sent to the rx:

CRSF::setSentSwitch(uint8_t index, uint8_t value)

Added calls to the above from tx_main.cpp to make it more obvious what was happening.